gatsby/plugins/gatsby-source-cks/src/api/api.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Low-level methods to access each of the CKS API endpoints
+ *
+ * @module
+ */
+
+import fetch from "node-fetch";
+import moment from "moment";
+import { Cache } from "gatsby";
+
+import {
+	ApiConfig,
+	ApiTopicsResponse,
+	ApiTopicResponse,
+	ApiTopicChangeResponse,
+	ApiPartialTopic,
+	ApiFullTopic,
+} from "./types";
+
+export const apiConfig: ApiConfig = {
+	apiBaseUrl: "",
+	apiKey: "",
+};
+
+export const configure = ({ apiBaseUrl, apiKey }: ApiConfig): void => {
+	apiConfig.apiBaseUrl = apiBaseUrl;
+	apiConfig.apiKey = apiKey;
+};
+
+const verifyConfig = (): true => {
+	if (
+		!apiConfig.apiBaseUrl ||
+		apiConfig.apiBaseUrl === "" ||
+		!apiConfig.apiKey ||
+		apiConfig.apiKey === ""
+	)
+		throw new Error(
+			"Please call configure with a base url and key before using the API."
+		);
+
+	return true;
+};
+
+/**
+ * Wrapper function around fetch that addds headers for
+ * authentication and JSON accept content type
+ */
+const fetchJson = async <TResponse>(path: string): Promise<TResponse> =>
+	verifyConfig() &&
+	(
+		await fetch(apiConfig.apiBaseUrl + path, {
+			headers: {
+				"ocp-apim-subscription-key": apiConfig.apiKey,
+				Accept: "application/json",
+			},
+		})
+	).json();
+
+/**
+ * Retrieves a list of all of the partial topics from the /topics API endpoint.
+ *
+ * @returns {Promise} A promise that resolves to an array of partial topics
+ */
+export const getAllPartialTopics = (): Promise<ApiTopicsResponse> =>
+	fetchJson("/topics");
+
+/**
+ * Gets a unique cache key for this version of the topic
+ */
+export const getTopicCacheKey = (partialTopic: ApiPartialTopic): string =>
+	`${partialTopic.id}_${partialTopic.currentVersion}`;
+
+/**
+ * Gets a full topic object from the /topic/id API endpoint.
+ * Uncached version of `getFullTopicCached`
+ *
+ * @see getFullTopicCached
+ */
+const getFullTopic = async (
+	partialTopic: ApiPartialTopic
+): Promise<ApiTopicResponse> =>
+	fetchJson<ApiTopicResponse>(partialTopic.currentVersionLink);
+
+/**
+ * Wrapper around the API method for retrieving a full topic.
+ * Looks in the Gatsby cache for that version of the topic
+ * before retrieving from the API.
+ *
+ * @see getFullTopic
+ *
+ * @returns A promise that resolves to a full topic
+ */
+export const getFullTopicCached = async (
+	partialTopic: ApiPartialTopic,
+	cache: Cache["cache"]
+): Promise<ApiFullTopic> => {
+	const topicCacheKey = getTopicCacheKey(partialTopic);
+
+	const cachedFullTopic = await (cache.get(
+		topicCacheKey
+	) as ApiFullTopic | null);
+
+	if (cachedFullTopic) return cachedFullTopic;
+
+	const apiFullTopic: ApiFullTopic = {
+		// Merge the partial topic properties in with the full one:
+		// there are a few properties on a partial topic the full one doesn't have
```

Comment:
I think  the name fulltopic could be misleading as partial topic contains properties that fulltopic doesnt meaning that it isnt "full".

Reply:
Ah but ApiFullTopic is 'full' because we merge the 2 together. I guess it's `getFullTopic` that's maybe misleading. Maybe it should just be `getSingleTopic`?